packages/firebase/externs/firebase-auth-externs.js

```diff
@@ -1416,7 +1507,59 @@ firebase.auth.Auth.prototype.signInAndRetrieveDataWithCredential = function(
 firebase.auth.Auth.prototype.signInWithCustomToken = function(token) {};
 
 /**
- * Asynchronously signs in using an email and password.
+ * Asynchronously signs in using an email and password and returns any additional
+ * user info data or credentials.
+ *
+ * Fails with an error if the email address and password do not match.
+ *
+ * Note: The user's password is NOT the password used to access the user's email
+ * account. The email address serves as a unique identifier for the user, and
+ * the password is used to access the user's account in your Firebase project.
+ *
+ * See also:
+ * {@link firebase.auth.Auth#createUserAndRetrieveDataWithEmailAndPassword}.
+ *
+ * <h4>Error Codes</h4>
+ * <dl>
+ * <dt>auth/invalid-email</dt>
+ * <dd>Thrown if the email address is not valid.</dd>
+ * <dt>auth/user-disabled</dt>
+ * <dd>Thrown if the user corresponding to the given email has been
+ *     disabled.</dd>
+ * <dt>auth/user-not-found</dt>
+ * <dd>Thrown if there is no user corresponding to the given email.</dd>
+ * <dt>auth/wrong-password</dt>
+ * <dd>Thrown if the password is invalid for the given email, or the account
+ *     corresponding to the email does not have a password set.</dd>
+ * </dl>
+ *
+ * @example
+ * firebase.auth().signInAndRetrieveDataWithEmailAndPassword(email, password)
+ *     .catch(function(error) {
+ *       // Handle Errors here.
+ *       var errorCode = error.code;
+ *       var errorMessage = error.message;
+ *       if (errorCode === 'auth/wrong-password') {
+ *         alert('Wrong password.');
+ *       } else {
+ *         alert(errorMessage);
+ *       }
+ *       console.log(error);
+ *     });
+ *
+ * @param {string} email The users email address.
+ * @param {string} password The users password.
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.signInAndRetrieveDataWithEmailAndPassword = function(
+  email,
+  password
+) {};
+
+/**
+ * Asynchronously signs in using an email and password. Will be depercated,
```

Comment:
This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#signInAndRetrieveDataWithEmailAndPassword}.